7 interface.md importance and examples (#103)
comp129/comp55: docs/labs/7-Interface.md

This is just coloring so that you are aware that you are dealing with an interface,
and that color will appear throughout for ```Emotional```.

> **Note:** While in this lab we're focusing on implementing our single ```Emotional``` interface,
> it's worth noting that one of the greatest strengths of interfaces in Java

> > is the ability to implement **multiple interfaces** within a single class.

Reviewer: ```suggestion
> is the *ability to implement **multiple interfaces** within a single class*.
```

> If you find yourself wanting to use more than one interface,

> > you can simply list them after the ```implements``` keyword,
> > separated by commas.

Reviewer: ```suggestion
> you can list their names after the ```implements``` keyword,
> separating them by commas.
```

> For example, if we wanted ```EmotionalOval``` not only to exhibit ```Emotional``` states

> > but also to respond to eating, we might also implement a ```Hungry``` interface like so:

Reviewer: ```suggestion
> but also to respond to cravings, we might also add a ```Hungry``` interface like so:
```

There is nothing in an interface about when it should be called or how it should behave.
Know that that it's a merely a collection of methods that one has to implement,
which could lead to some great examples of polymorphism.
This flexibility is what makes interfaces so powerful,

> and learning how to make them work for you will give you a lot of power when you start your final projects.

Reviewer: ```suggestion
and ***learning how to make interfaces work for you
will give you a lot of power*** when you start your final projects.
```
